crates/forge/src/coverage.rs

```diff
@@ -77,66 +77,68 @@ fn format_cell(hits: usize, total: usize) -> Cell {
     cell
 }
 
+/// Writes the coverage report in [LCOV]'s [tracefile format].
+///
+/// [LCOV]: https://github.com/linux-test-project/lcov
+/// [tracefile format]: https://man.archlinux.org/man/geninfo.1.en#TRACEFILE_FORMAT
 pub struct LcovReporter<'a> {
-    /// Destination buffer
-    destination: &'a mut (dyn Write + 'a),
+    out: &'a mut (dyn Write + 'a),
 }
 
 impl<'a> LcovReporter<'a> {
-    pub fn new(destination: &'a mut (dyn Write + 'a)) -> Self {
-        Self { destination }
+    /// Create a new LCOV reporter.
+    pub fn new(out: &'a mut (dyn Write + 'a)) -> Self {
+        Self { out }
     }
 }
 
 impl CoverageReporter for LcovReporter<'_> {
     fn report(self, report: &CoverageReport) -> eyre::Result<()> {
-        for (file, items) in report.items_by_source() {
-            let summary = items.iter().fold(CoverageSummary::default(), |mut summary, item| {
-                summary += item;
-                summary
-            });
+        for (path, items) in report.items_by_file() {
+            let summary = CoverageSummary::from_items(items.iter().copied());
 
-            writeln!(self.destination, "TN:")?;
-            writeln!(self.destination, "SF:{}", file.display())?;
+            writeln!(self.out, "TN:")?;
+            writeln!(self.out, "SF:{}", path.display())?;
 
             for item in items {
-                let line = item.loc.line;
+                let line = item.loc.lines.start;
+                let line_end = item.loc.lines.end - 1;
                 let hits = item.hits;
                 match item.kind {
-                    CoverageItemKind::Function { name } => {
+                    CoverageItemKind::Function { ref name } => {
                         let name = format!("{}.{name}", item.loc.contract_name);
-                        writeln!(self.destination, "FN:{line},{name}")?;
-                        writeln!(self.destination, "FNDA:{hits},{name}")?;
+                        writeln!(self.out, "FN:{line},{line_end},{name}")?;
```

Comment:
that's awesome, we can get rid of this from book after merged https://book.getfoundry.sh/reference/forge/forge-coverage#description